coreos-koji-tagger/coreos_koji_tagger.py

```diff
@@ -264,69 +288,111 @@ def __call__(self, message: fedora_messaging.api.Message):
         # koji builds that aren't in the tag we can add the koji pkg to the
         # tag (if needed) and then tag the koji build into the tag.
 
-        # parse the lockfile and get a set of rpm NEVRAs
+        # parse the lockfile and get a set of rpm NEVRAs (strings)
         desiredrpms = set(parse_lockfile_data(r.text))
 
-        # convert the rpm NEVRAs into a list of srpm NVRA (format of koji
-        # build name)
-        buildsinfo = get_buildsinfo_from_rpmnevras(desiredrpms)
-        desiredbuilds = set(buildsinfo.keys())
+        # convert the NEVRAs into a dict of build IDs -> BuildInfo objects
+        buildsinfo = self.get_buildsinfo_from_rpmnevras(desiredrpms)
+        desiredbuildids = buildsinfo.keys()
 
-        # Grab the list of pkgs that can be tagged into the tag
-        pkgsintag = get_pkgs_in_tag(self.target_tag)
+        # Grab the build IDs of currently tagged builds
+        currentbuildids = self.get_tagged_buildids(self.target_tag)
 
-        # Grab the currently tagged builds and convert it into a set
-        currentbuilds = set(get_tagged_builds(self.target_tag))
+        # Get the set of names of pkgs needed
+        desiredpkgs = set([x.name for x in buildsinfo.values()])
 
-        # Find out the difference between the current set of builds
-        # that exist in the koji tag and the desired set of builds to
-        # be added to the koji tag.
-        buildstotag = list(desiredbuilds.difference(currentbuilds))
+        # Grab the set of names of pkgs that can be tagged into the tag
```

Comment:
Tagging things into a tag doesn't really make the most sense from the outside perspective but I think that might be outside the scope of this review :)

Reply:
:)